Cython/Compiler/Parsing.py

```diff
@@ -318,7 +318,8 @@ def p_typecast(s):
     is_memslice = isinstance(base_type, Nodes.MemoryViewSliceTypeNode)
     is_template = isinstance(base_type, Nodes.TemplatedTypeNode)
     is_const_volatile = isinstance(base_type, Nodes.CConstOrVolatileTypeNode)
-    if not is_memslice and not is_template and not is_const_volatile and base_type.name is None:
+    is_ctuple = isinstance(base_type, Nodes.CTupleBaseTypeNode)
+    if not is_memslice and not is_template and not is_const_volatile and not is_ctuple and base_type.name is None:
```

Comment:
I think this has become long enough to start making it easier to read.
```suggestion
    if not (is_memslice or is_template or is_const_volatile or is_ctuple) and base_type.name is None:
```
And given that, I wonder if it's not better to just say
```cython
isinstance(base_type, (
    list,
    exceptional,
    types,
    here,
))
```